### DISCUSSION TITLE
Scenes/Groups/Associations: Store Natively in Device?

### DISCUSSION THREAD

**Discussion by ccclapp** [home-assistant/architecture#946] [Entity Models]
My understanding may be mistaken, but I understand that Insteon Scenes, Zigbee Groups and Z-Wave Associations (grouped devices for each protocol) can be written to and stored in the devices, natively, but that HA has chosen not to implement that.

For example, in the case of multiple Insteon, e.g. switches, in a scene written to natively: each switch listens for 2 commands, a. "Turn myself on" and b. "Turn on if Scene 1 is told to turn on" (this includes the devices dim level etc).

This has 2 great benefits:
1. a scene command is a single message: "Turn on Scene 1", NOT separate massages to every member of the scene
2. . because these are written natively to the devices, if a controller/plm etc is offline/broken, the scene is still fully functional from the controlling switch(s).

My understanding is HA DOES NOT utilize this native device-writing capability and instead keeps all communication strictly between itself and devices.  Therefore, when controlling an Insteon Scene (Zig Group, Z-W Association): 10 separate commands would be sent if 10 devices in the scene, and the scene would not work if the plm/controller was offline etc.

I gather this is why e.g. for Insteon, users utilize UD ISY (eisy), which does write natively.  I imagine ditto for alternate zigbee and z-wave hubs.  

It would be fantastic if the HA architecture would enable natively writing to devices, so folks could keep everything inside HA and not rely on multiple external hubs and corresponding intermediary integrations to connect to HA (usually with some meaningful loss of functionality and added complexity.

If the above is fundamentally correct, is there a reason for maintaining this architecture?

I am trying to better understand and map a direction, accordingly...I would love to keep HA as the single point of truth!

Thanks in advance and for all the hard work you all do!

**Comment by frenck** [comment]
I think you misunderstood what this repository is for.

It seems you are having questions about the choices Home Assistant made, for which we have our community forums and Discord chat. In case this is a feature request, please use the community forum, we have a dedicated feature request section there.

This repository is for proposal architectural changes that require approval. These proposals are made by the implementers and have a technical nature. Your post does seem to fit that.

Closing this discussion for that reason.

../Frenck
